<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Data input, cleaning and pre-processing &mdash; PyWGCNA  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/graphviz.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> PyWGCNA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../README.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../README.html#tutorials">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules.html">API Documentation</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">PyWGCNA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Data input, cleaning and pre-processing</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/tutorials/Data_format.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="data-input-cleaning-and-pre-processing">
<h1>Data input, cleaning and pre-processing<a class="headerlink" href="#data-input-cleaning-and-pre-processing" title="Permalink to this headline"></a></h1>
<p>This is the first step of any network analysis.
We show here how to load typical expression data,
pre-process them into a format suitable for
network analysis, and clean the data by removing
obvious outlier samples as well as genes and
samples with excessive numbers of missing entries.</p>
<ul class="simple">
<li><p><span class="xref myst">Data Input</span></p>
<ul>
<li><p><span class="xref myst">anndata format</span></p></li>
<li><p><span class="xref myst">separate format</span></p>
<ul>
<li><p><span class="xref myst">Gene Expression</span></p></li>
<li><p><span class="xref myst">Gene Information</span></p></li>
<li><p><span class="xref myst">Sample Information</span></p></li>
<li><p><span class="xref myst">Other parameters</span></p></li>
</ul>
</li>
</ul>
</li>
<li><p><span class="xref myst">Data cleaning and pre-processing</span></p></li>
</ul>
<section id="data-input">
<h2>Data Input<a class="headerlink" href="#data-input" title="Permalink to this headline"></a></h2>
<p>We store <strong>raw</strong> expression data along information in <a class="reference external" href="https://anndata.readthedocs.io/en/latest/">anndata</a> format in <code class="docutils literal notranslate"><span class="pre">geneExpr</span></code> variable.
you can pass your expression data, gene and sample information all together or separately:</p>
<section id="expression-data-gene-and-sample-information-all-together-in-anndata-format">
<h3>expression data, gene and sample information all together in anndata format<a class="headerlink" href="#expression-data-gene-and-sample-information-all-together-in-anndata-format" title="Permalink to this headline"></a></h3>
<p>If you already have your expression data in anndata format you can define your pyWGCNA object by passing your variable in <code class="docutils literal notranslate"><span class="pre">anndata</span></code> format.
keep in mind X should be expression matrix. var is gene information and obs is sample information.</p>
</section>
<section id="expression-data-gene-and-sample-information-separately">
<h3>expression data, gene and sample information separately<a class="headerlink" href="#expression-data-gene-and-sample-information-separately" title="Permalink to this headline"></a></h3>
<p>you can pass the paths that store each information or the table contains them.</p>
<section id="gene-expression">
<h4>Gene Expression<a class="headerlink" href="#gene-expression" title="Permalink to this headline"></a></h4>
<p>The expression data is a table which the rows are samples and columns are genes.
the first column is going to be sample id or sample name and first column should be gene id or gene name which both of them which should be unique.</p>
<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>gene_id</th>
      <th>sample_11615</th>
      <th>sample_11616</th>
      <th>sample_11617</th>
      <th>sample_11618</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>ENSMUSG00000000001.4</td>
      <th>12.04</th>
      <th>11.56</th>
      <th>16.06</th>
      <th>13.18</th>
    </tr>
    <tr>
      <td>ENSMUSG00000000028.15</td>
      <th>1.35</th>
      <th>1.63</th>
      <th>1.28</th>
      <th>1</th>
    </tr>
  </tbody>
</table>
</div>
</section>
<section id="gene-information">
<h4>Gene Information<a class="headerlink" href="#gene-information" title="Permalink to this headline"></a></h4>
<p>The gene information is a table which contains additional information about each genes.
It should have a same order as gene expression matrix.</p>
<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>gene_id</th>
      <th>gene_name</th>
      <th>gene_type</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>ENSMUSG00000000001.4</td>
      <th>Gnai3</th>
      <th>Protein coding</th>
    </tr>
    <tr>
      <td>ENSMUSG00000000028.15</td>
      <th>Cdc45</th>
      <th>Protein coding</th>
    </tr>
  </tbody>
</table>
</div>
</section>
<section id="sample-information">
<h4>Sample Information<a class="headerlink" href="#sample-information" title="Permalink to this headline"></a></h4>
<p>The sample information is a table which contains additional information about each sample.
It should have a same order as gene expression matrix.</p>
<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>Sample_id</th>
      <th>Sex</th>
      <th>Age</th>
      <th>Genotype</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>sample_11615</td>
      <td>Female</td>
      <td>4 month</td>
      <td>5xFAD</td>
    </tr>
    <tr>
      <td>sample_11616</td>
      <td>Male</td>
      <td>4 month</td>
      <td>5xFAD</td>
    </tr>
    <tr>
      <td>sample_11617</td>
      <td>Female</td>
      <td>4 month</td>
      <td>BL6</td>
    </tr>
    <tr>
      <td>sample_11618</td>
      <td>Male</td>
      <td>4 month</td>
      <td>BL6</td>
    </tr>
  </tbody>
</table>
</div>
</section>
</section>
<section id="other-parameters">
<h3>Other parameters<a class="headerlink" href="#other-parameters" title="Permalink to this headline"></a></h3>
<p>These are other parameters we suggest checking them before starting any analysis.</p>
<ul class="simple">
<li><p><strong>name</strong>: name of the WGCNA we used to visualize data (default: <code class="docutils literal notranslate"><span class="pre">WGCNA</span></code>)</p></li>
<li><p><strong>save</strong>: define whether you want to save result of important steps or not (If you want to set it
<code class="docutils literal notranslate"><span class="pre">TRUE</span></code> you should have a write access on the output directory)</p></li>
<li><p><strong>outputPath</strong>: define where you want to save your data, otherwise it will be store near the code.</p></li>
<li><p><strong>TPMcutoff</strong>: cut off for removing genes that expressed under this number along samples</p></li>
<li><p><strong>networkType</strong> : Type of networks (default: <code class="docutils literal notranslate"><span class="pre">signed</span> <span class="pre">hybrid</span></code> and Options: <code class="docutils literal notranslate"><span class="pre">unsigned</span></code>, <code class="docutils literal notranslate"><span class="pre">signed</span></code> and <code class="docutils literal notranslate"><span class="pre">signed</span> <span class="pre">hybrid</span></code>)</p></li>
<li><p><strong>adjacencyType</strong>: Type of adjacency matrix (default: <code class="docutils literal notranslate"><span class="pre">signed</span> <span class="pre">hybrid</span></code> and Options: <code class="docutils literal notranslate"><span class="pre">unsigned</span></code>, <code class="docutils literal notranslate"><span class="pre">signed</span></code> and <code class="docutils literal notranslate"><span class="pre">signed</span> <span class="pre">hybrid</span></code>)</p></li>
<li><p><strong>TOMType</strong>: Type of topological overlap matrix(TOM) (default: <code class="docutils literal notranslate"><span class="pre">signed</span></code> and Options: <code class="docutils literal notranslate"><span class="pre">unsigned</span></code> and <code class="docutils literal notranslate"><span class="pre">signed</span></code>)</p></li>
</ul>
</section>
</section>
<section id="data-cleaning-and-pre-processing">
<h2>Data cleaning and pre-processing<a class="headerlink" href="#data-cleaning-and-pre-processing" title="Permalink to this headline"></a></h2>
<p>PyWGCNA checks data for genes and samples with too many missing values.</p>
<ol class="arabic simple">
<li><p>Remove genes without any expression more than <code class="docutils literal notranslate"><span class="pre">TPMcutoff</span></code> value (default one) across all samples.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">goodSamplesGenes()</span></code> function to find genes and samples with too many missing values.</p></li>
<li><p>Cluster the samples (use <a class="reference external" href="https://docs.scipy.org/doc/scipy/reference/cluster.hierarchy.html#module-scipy.cluster.hierarchy">Hierarchical clustering</a>
from <a class="reference external" href="https://scipy.org/">scipy</a>) to see if there are any obvious outliers.
you can define value the height by <code class="docutils literal notranslate"><span class="pre">cut</span></code> value. By default, we don’t remove any sample by hierarchical clustering</p></li>
</ol>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Narges Rezaie.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>